bpftimetool: fixes unnecessary report of shm removed when there's nothing to remove

aneeshdamle11 opened the pull request:
<!--# Pull Request Template-->

## Description

<!--Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.-->
bpftimetool reports "Global shm removed" even if there is nothing to remove. Fixed here.

Fixes #169

## Type of change

<!--Please delete options that are not relevant.-->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration-->

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

Officeyutong commented on runtime/src/bpftime_shm_internal.cpp:
```diff
@@ -41,9 +41,11 @@ extern "C" void bpftime_destroy_global_shm()
-	SPDLOG_INFO("Global shm removed");
+	const char *shm_name = get_global_shm_name();
+	boost::interprocess::shared_memory_object::remove(shm_name);
+	if (shm_name != DEFAULT_GLOBAL_SHM_NAME) {
```
Why compare c-string in this way?

aneeshdamle11 replied:
I'll correct it in the next commit with strcmp

yunwei37 commented:
See https://www.boost.org/doc/libs/1_80_0/doc/html/interprocess/sharedmemorybetweenprocesses.html#interprocess.sharedmemorybetweenprocesses.sharedmemory.removing

Does that mean you should compare the `shm_name` pointer with nullptr? What is the `shm_name` pointer after the sharedmemory is successfully removed?

aneeshdamle11 commented:
I looked in the code further and pored over my approach. bpftime_remove_global_shm_name() should remove the shared memory segment from its system.
It gets the name of the global shared memory from get_global_shm() which fetches the value of an environment variable BPFTIME_GLOBAL_SHM_NAME.
I am a bit confused on how to test it properly. Could you help with any inputs?

aneeshdamle11 commented:
According to the referred doc: https://www.boost.org/doc/libs/1_80_0/doc/html/interprocess/sharedmemorybetweenprocesses.html#interprocess.sharedmemorybetweenprocesses.sharedmemory.removing, I changed the remove function to check it's return value. I realised I didn't need the shm_name variable to do this.

I tested the bpftimetool assigning a shared memory using the load function. Then I removed it twice, the second time showing no output.
![image](https://github.com/eunomia-bpf/bpftime/assets/79093385/4812c92f-c126-424d-b2ed-c72f9d9959ca)

Does this test work? I am a bit new code, but tried my best understanding this part. Please provide any inputs if my understanding needs to be improved.

Officeyutong reviewed (approved):
Thanks for your work!

Officeyutong commented:
> According to the referred doc: https://www.boost.org/doc/libs/1_80_0/doc/html/interprocess/sharedmemorybetweenprocesses.html#interprocess.sharedmemorybetweenprocesses.sharedmemory.removing, I changed the remove function to check it's return value. I realised I didn't need the shm_name variable to do this.
> 
> I tested the bpftimetool assigning a shared memory using the load function. Then I removed it twice, the second time showing no output. ![image](https://private-user-images.githubusercontent.com/79093385/309256704-4812c92f-c126-424d-b2ed-c72f9d9959ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyOTc2NjcsIm5iZiI6MTcwOTI5NzM2NywicGF0aCI6Ii83OTA5MzM4NS8zMDkyNTY3MDQtNDgxMmM5MmYtYzEyNi00MjRkLWIyZWQtYzcyZjlkOTk1OWNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAxVDEyNDkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxYTE3M2JhNzg4NTk1YWU3YmI1YzA4YjBmNWJiZTA0ODQ4NjYzODkxMDM2MGYzYzE3ZWYzNTEwMDNjNDA1MWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZLuNdfJbuRQOYTD1XU0UgWyMwx2A4xLuiMxpab34r-s)
> 
> Does this test work? I am a bit new code, but tried my best understanding this part. Please provide any inputs if my understanding needs to be improved.

I think it's correst. I'll merge this PR once the CI passed

aneeshdamle11 commented:
Glad I could help! And thank you for giving me this opportunity to contribute!
